### PR TITLE
Emitter gun mounts

### PR DESCRIPTION
**Pull request by Time-Green**
Simple PR, allows you to mount energy guns into emitters. You can now insert anything from a taser to an x-ray gun, and have the emitter shoot that round.

:cl: Time-Green
add: You can now mount energy guns into emitters
fix: portal guns no longer runtime when fired by turrets
/:cl:

Turrets already do exactly this, and why not let people do it with emitters

### CODE REVIEW

**Comment by Qustinnus** [comment]
would work better as its own object imo

**Comment by Kagus** [comment]
Worth noting that energy/gun also includes fun things like the particle acceleration beam rifle, the portal gun, and the gravity gun.

I'm not sure how any of those things work when fired by a non-human.

**Comment by 81Denton** [comment]
Can you still add assemblies (timer, remote signaler etc) to the emitter wire when something is mounted?

**Comment by Time-Green** [comment]
All this does is change the projectile type and sound, so yes

**Comment by Grazyn** [comment]
RIP blob

**Comment by GravitronBlue** [comment]
Unless there's already a specific check for the portal gun, this PR sounds like it has potential to get a set(s) of portals connecting the station to the Centcom z-level. I'd ask that you test whether this is feasible.

**Comment by Tlaltecuhtli** [comment]
its the micro laser you use to make the emitter

**Comment by Time-Green** [comment]
@GravitronBlue teleporting has been disabled on centcom for ages now

**Comment by Time-Green** [comment]
Right, portal guns dont work, wich I kinda expected, but it also caused a fucking runtime because qdelling is async. 
Anyway that's fixed now.

**Comment by AnturK** [comment]
A lot of projectiles require special setup. Use the turret properties instead. These are required to work out of the box.

**Comment by Time-Green** [comment]
https://github.com/tgstation/tgstation/blob/d80ae38e1e96f7f95d8fc94868744c5bd06718f1/code/game/machinery/porta_turret/portable_turret.dm#L550
Uses the same stuff. Wormhole projector was runtiming like crazy on the turret, wich is what I fixed.
I tested some other exotic lasers, and apart from particle rifle not doing damage, it all works fine now.

**Comment by AnturK** [comment]
That's done after it already received proper projectile through get_turret_properties. As opposed to directly grabbing the projectile type.

**Comment by Time-Green** [comment]
Right, it now uses get_turret_properties. There's a new wire that switches the projectile type it fires, if there's more then one

File: code/modules/projectiles/projectile/special/wormhole.dm [hunk 1]
```diff
@@ -20,7 +20,8 @@
 	if(casing)
 		gun = casing.gun
 
+
 /obj/item/projectile/beam/wormhole/on_hit(atom/target)
 	if(!gun)
-		qdel(src)
+		return qdel(src)
```
Comment (AnturK) [inline code/modules/projectiles/projectile/special/wormhole.dm]: qdel does not have any meaningful return value, so this should be on next line to not confuse people reading this later.

**Review by AnturK — CHANGES_REQUESTED** [review]
Looks good now, just a nitpick.

**Review by AnturK — DISMISSED** [review]

**Comment by Time-Green** [comment]
I found a bug while reading code, that’d break shit when you would click it with a nodrop gun.

**Pull request merged**